src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -0,0 +1,163 @@
+            return false;
+        }
+    }
+
+    private boolean verifyCourse(teammates.storage.sqlentity.Course sqlEntity, Course datastoreEntity) {
+        return sqlEntity.getId().equals(datastoreEntity.getUniqueId())
+                && sqlEntity.getName().equals(datastoreEntity.getName())
+                && sqlEntity.getTimeZone().equals(datastoreEntity.getTimeZone())
+                && sqlEntity.getInstitute().equals(datastoreEntity.getInstitute())
+                && sqlEntity.getCreatedAt().equals(datastoreEntity.getCreatedAt())
+                && datastoreEntity.getDeletedAt() == null ? sqlEntity.getDeletedAt() == null
+                        : sqlEntity.getDeletedAt().equals(datastoreEntity.getDeletedAt());
+    }
+
+    private boolean verifySectionChain(teammates.storage.sqlentity.Course sqlEntity) {
+        // Get old and new students
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", sqlEntity.getId())
+                .list();
+        List<Student> newStudents = getNewStudents(sqlEntity.getId());
+
+        // Group students by section
+        Map<String, List<CourseStudent>> sectionToOldStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+        Map<String, List<Student>> sectionToNewStuMap = newStudents.stream()
+                .collect(Collectors.groupingBy(Student::getSectionName));
+
+        List<Section> newSection = sqlEntity.getSections();
+
+        boolean isSectionsCountEqual = newSection.size() != sectionToOldStuMap.size()
+                || newSection.size() != sectionToNewStuMap.size();
+        if (!isSectionsCountEqual) {
+            return false;
+        }
+
+        return newSection.stream().allMatch(section -> {
+            List<CourseStudent> oldSectionStudents = sectionToOldStuMap.get(section.getName());
+            List<Student> newSectionStudents = sectionToNewStuMap.get(section.getName());
+
+            // If sectionStudent is null, then section is not present in sql
+            boolean isSectionNamePresent = oldSectionStudents != null && newSectionStudents != null;
+            if (!isSectionNamePresent) {
+                return false;
+            }
+
+            // Group students by team
+            Map<String, List<CourseStudent>> teamNameToOldStuMap = oldSectionStudents.stream()
+                    .collect(Collectors.groupingBy(CourseStudent::getTeamName));
+            Map<String, List<Student>> teamNameToNewStuMap = newSectionStudents.stream()
+                    .collect(Collectors.groupingBy(Student::getTeamName));
+            return verifyTeams(section, teamNameToOldStuMap, teamNameToNewStuMap);
+        });
+
+    }
+
+    private boolean verifyTeams(Section newSection,
+            Map<String, List<CourseStudent>> teamNameToOldStuMap, Map<String, List<Student>> teamNameToNewStuMap) {
+
+        List<Team> newTeam = newSection.getTeams();
```

Comment:
Small nit: `newTeams` instead of `newTeam`. Similarly above, `newSections` instead of `newSection`

Reply:
Good catch!